File: go/avoid_flag_arguments.go
// Avoid Flag Arguments
// Avoid providing boolean arguments (true/false) to functions or methods.
// You could pass a string with a clearer name (for example).

package main

import "fmt"

// Before
func greetBefore(name string, formal bool) {
	if formal {
		fmt.Printf("Good evening, %s.\n", name)
	} else {
		fmt.Printf("Hi, %s!\n", name)
	}
}

// After
func greet(name, formality string) {
	switch formality {
	case "formal":
		fmt.Printf("Good evening, %s.\n", name)
	case "informal":
		fmt.Printf("Hi, %s!\n", name)
	}
}

func main() {
	greetBefore("Alice", true)
	greetBefore("Bob", false)

	fmt.Println()

	greet("Alice", "formal")
	greet("Bob", "informal")
}

File: javascript/avoid_flag_arguments.js
// Avoid Flag Arguments
// Avoid providing boolean arguments (true/false) to functions or methods.
// You could pass a string with a clearer name (for example).

// Before
function greet(name, formal) {
  if (formal) {
    console.log(`Good evening, ${name}.`);
  } else {
    console.log(`Hi, ${name}!`);
  }
}

greet("Alice", true);
greet("Bob", false);


// After
function greet(name, formality) {
  if (formality === "formal") {
    console.log(`Good evening, ${name}.`);
  } else if (formality === "informal") {
    console.log(`Hi, ${name}!`);
  }
}

greet("Alice", "formal");
greet("Bob", "informal");
